### ISSUE TITLE
Unificar las conversaciones?

### ISSUE THREAD

**Issue by deivid-rodriguez** [alabs/nolotiro.org#130]
Creo que debería haber una única conversación entre 2 usuarios (como en Telegram o Whatsaap). A veces un usuario con el que estoy conversando me abre una nueva conversación y te pierdes un poco si no puedes leer el contexto.

**Comment by andreslucena** [comment]
+1, aunque creo que la unificación debería partir del anuncio más que de la unión entre dos usuarios, como lo hacen en Wallapop, la has usado alguna vez?

**Comment by deivid-rodriguez** [comment]
No, pero entiendo lo que dices. También valdría. De hecho, se podrían añadir en algún lado los datos del anuncio del que se está conversando.

Wallapop nunca lo he usado, pero me voy a registrar que tengo unas cosillas que vender. :)

**Comment by deivid-rodriguez** [comment]
Bueno, algunas reflexiones sobre esto.

- Creo que ahora mismo, desde el punto de vista del regalador, este es uno de los puntos más débiles de nolotiro. Gestionar regalos es un auténtico caos.

- He estado viendo cómo se hace esto en wallapop. Efectivamente, está guay. Tienes un link al anuncio al principio de la conversación y las conversaciones se agrupan por usuario + anuncio También puedes acceder fácilmente al usuario con el que estás hablando (aunque esto podría mejorarse en wallapop también).

- Esta feature requiere un rediseño. Habría que convocar a Eneko por estos lares quizá.

- Se me ocurre un first step que no requeriría rediseño y que podría facilitar mucho las cosas. Unificar a nivel de usuario, y cuando se contacta a un usuario desde un anuncio concreto, simplemente añadir un link a ese anuncio junto al mensaje. En plan... "en relación a + link" o algo así.

Thoughts?

**Comment by deivid-rodriguez** [comment]
En realidad no hace falta rediseño tampoco para "hacerlo bien". De momento voy a unificar por usuario + anuncio y siemplemente enlazo al anuncio en el título de la conversación. Os parece?

**Comment by andreslucena** [comment]
:+1:

**Comment by deivid-rodriguez** [comment]
Le he dado otra vuelta al tema y ya no tengo claro que unificar por regalo sea la mejor opción.

En nolotiro creo que es mucho más habitual tener muchas conversaciones con la misma usuaria. Cuando se regalan varias cosas, es muy común que la misma usuaria te pida más de una.

En general, en estos casos se acaba casi siempre conversando en un único hilo, quedando el resto limitados a "hablamos por la otra conversación". Vamos, que no acabo de ver que unificar por regalo sea práctico en nolotiro. Creo que una única conversación por usuario donde se vayan añadiendo links a los regalos que se mencionen funcionaría bien.

**Comment by andreslucena** [comment]
Entiendo el problema que planteas, lo que si agrega complejidad de cara a la interfaz que el usuario hace para relacionar el mensaje con los anuncios. Seria el propio link que copia y pega en el mensaje?

**Comment by deivid-rodriguez** [comment]
Mi idea es enlazar ciertos mensajes específicos de la conversación con anuncios (cuando se crear desde "enviar un mensaje privado al anunciante" de un anuncio). En esos caso, simplemente mostrar un link al anuncio por encima del mensaje. Por ejemplo,

En relación a [Cafetera Italiana] (#):

/------------------------------------------------
| Y debajo el globito con el mensaje.
\ -------------------------------------------------

**Comment by andreslucena** [comment]
OK, creo que entiendo la UI, por mi bien. Tiene logica que las conversaciones vayan por la relación con anunciantes, asi si alguien te pide muchas cosas lo tenes en el mismo historial de conversacion. El problema puede ser de performance en el futuro, que puede haber cientos de mensajes en la misma conversa y habría que pensar un metodo de paginación o similar.

**Comment by deivid-rodriguez** [comment]
Correcto, hay que ser cuidadosas con esto. Yo tengo muchas ideas con respecto a este tema, ya os las comentaré cuando hablemos por videoconf!
